assets/js/components/PostSearcherAutoSuggest.js

```diff
@@ -154,16 +155,23 @@ export default function PostSearcherAutoSuggest( {
 				typeof AbortController === 'undefined'
 					? undefined
 					: new AbortController();
-			API.get(
+
+			const fetchPromise = API.get(
 				'core',
 				'search',
 				'post-search',
 				{ query: encodeURIComponent( debouncedValue ) },
 				{ useCache: false, signal: controller?.signal }
 			)
 				.then( ( res ) => setResults( res ) )
-				.catch( () => setResults( [] ) )
-				.finally( () => setIsLoading?.( false ) );
+				.catch( () => setResults( null ) );
+
+			lastfetchRequest.current = fetchPromise;
+			fetchPromise.finally( () => {
+				if ( fetchPromise === lastfetchRequest.current ) {
+					setIsLoading?.( false );
+				}
+			} );
```

Comment:
Any reason we can't do `try / catch / finally` here instead of the promise chaining? Especially as we're defining the flow in this separated way, I find it a bit harder to follow the logic.